mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -604,6 +604,21 @@ def delete_tag(self, run_id, key):
                     error_code=INVALID_STATE)
             session.delete(filtered_tags[0])
 
+    def list_all_columns(self, experiment_id, run_view_type):
+        stages = set(LifecycleStage.view_type_to_stages(run_view_type))
+        with self.ManagedSessionMaker() as session:
+            tags = [r[0] for r in session.query(SqlTag.key).join(SqlRun)
+                    .filter(SqlRun.experiment_id == experiment_id,
+                            SqlRun.lifecycle_stage.in_(stages)).distinct().all()]
+            metrics = [r[0] for r in session.query(SqlLatestMetric.key).join(SqlRun)
+                       .filter(SqlRun.experiment_id == experiment_id,
+                               SqlRun.lifecycle_stage.in_(stages)).distinct().all()]
+            params = [r[0] for r in session.query(SqlParam.key).join(SqlRun)
+                      .filter(SqlRun.experiment_id == experiment_id,
+                              SqlRun.lifecycle_stage.in_(stages)).distinct().all()]
```

Comment:
These queries can be super expensive if the experiment has a lot of runs or runs have lots of params/tags/... etc

Reply:
these queries are not actually expensive compared to what search runs is doing.
search runs today doesn't scale beyond 2000 runs and 100 of params/tags/metrics
These queries definitely scale way beyond that as they do not contain any join, so it's definitely in the range of scale that mlflow currently support